packages/angular/src/migrations/update-13-8-4/migrate-karma-conf.ts

```diff
@@ -0,0 +1,60 @@
+import type { Tree } from '@nrwl/devkit';
+import {
+  addDependenciesToPackageJson,
+  joinPathFragments,
+  readProjectConfiguration,
+  formatFiles,
+} from '@nrwl/devkit';
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+export default async function (tree: Tree) {
+  const executorName = '@angular-devkit/build-angular:karma';
+
+  forEachExecutorOptions(tree, executorName, ({ karmaConfig }, projectName) => {
+    const project = readProjectConfiguration(tree, projectName);
+
+    const pathToKarma = joinPathFragments(project.root, karmaConfig);
+    if (!tree.exists(pathToKarma)) {
+      return;
+    }
+
+    const karmaContents = tree.read(pathToKarma, 'utf-8');
+    const updatedKarmaContents = karmaContents.replace(
+      /coverageIstanbulReporter/g,
+      'coverageReporter'
+    );
+
+    tree.write(pathToKarma, updatedKarmaContents);
+  });
+
+  const pathToRootKarmaConf = 'karma.conf.js';
+  if (!tree.exists(pathToRootKarmaConf)) {
+    return;
+  }
+
+  const rootKarmaContents = tree.read(pathToRootKarmaConf, 'utf-8');
+  const updatedKarmaContents = rootKarmaContents
+    .replace(/coverageIstanbulReporter/g, 'coverageReporter')
+    .replace(/karma-coverage-istanbul-reporter/, 'karma-coverage')
+    .replace(
+      /reports: \['html', 'lcovonly'\]/,
+      `subdir: '.',\nreporters: [{ type: 'html' }, { type: 'text-summary' }]`
+    )
+    .replace(/'\.\.\/\.\.\/coverage'/, `'./coverage'`);
```

Comment:
I think this can be problematic. Consider this scenario:

```js
coverageIstanbulReporter: {
  dir: join(__dirname, '../../coverage'),
  reports: ['html', 'lcovonly', 'text-summary'],
  fixWebpackSourcePaths: true,
},
```

The current logic will correctly replace `coverageIstanbulReporter` with `coverageReporter` but it won't replace the `reports` key as the dev had an additional reporter (or could have one less or the same number but changed). The `fixWebpackSourcePaths` also didn't get removed. The workspace ends up with a wrong config for the coverage.

Reply:
True, i guess it would be better to just replace all the reporters in `reports` with the two we know for definite are supported. My fear was trying to migrate ones that weren't supported by `karma-coverage`

also, good catch